src/core/Akka.Persistence.Tests/PersistentActorSpec.cs

```diff
@@ -611,5 +611,21 @@ public void PersistentActor_should_brecover_the_message_which_caused_the_restart
             persistentActor.Tell("boom");
             ExpectMsg("failed with TestException while processing boom");
         }
+
+        [Fact]
+        public void PersistentActor_should_be_able_to_persist_events_that_happen_during_recovery()
+        {
+            var persistentActor = ActorOf(Props.Create(() => new PersistInRecovery(Name)));
+            persistentActor.Tell(GetState.Instance);
+            ExpectMsgInOrder("a-1", "a-2", "rc-1", "rc-2");
+            persistentActor.Tell(GetState.Instance);
+            ExpectMsgInOrder("a-1", "a-2", "rc-1", "rc-2", "rc-3");
```

Comment:
There's some race condition (at least in the JVM) between the GetState and the PersistAsync, so they've simplified the test like this:

```expectMsgAnyOf(List("a-1", "a-2", "rc-1", "rc-2"), List("a-1", "a-2", "rc-1", "rc-2", "rc-3"))```

But I'm not sure this is possible with the current ExpectMsgAnyOf in Akka.NET

Reply:
@Aaronontheweb about this?

Reply:
I saw the comment - the test passes currently, right? If it's going to be racy, do we need to implement some new features in the `TestKit` in order to do this correctly? What's the issue?

Reply:
The test runs fine, at least every time I've ran it. But I noticed they (JVM) changed it since the original PR for the reason I mentioned, just so you know.